mobile/ios/ios-core.ts

```diff
@@ -669,15 +670,26 @@ class WinSocket implements Mobile.IiOSDeviceSocket {
 		return data;
 	}
 
-	public readSystemLog(printData: any) {
+	public readSystemLogBlocking() {
 		let data = this.read(WinSocket.BYTES_TO_READ);
 		while (data) {
-			printData(data);
+			let output = ref.readCString(data, 0);
+			process.send(output);
 			data = this.read(WinSocket.BYTES_TO_READ);
 		}
 		this.close();
 	}
 
+	public readSystemLog(printData: any) {
+		let ch = require("child_process");
```

Comment:
We can think about a mockable dependency here instead of requiring node's child_process module directly